### ISSUE TITLE
I can't view paragraphs made up of more than one sentence correctly in PDF. In the PDF I can only view part of the first line.

### ISSUE THREAD

**Issue by me-suzy** [PyFPDF/fpdf2#486]
hello, I can't view paragraphs made up of more than one sentence correctly in PDF. 
In the PDF I can only view part of the first line from the paragraph. See this print screen:

![](https://snipboard.io/PQ1n7j.jpg)

This is the paragraph:

`<p class="text_obisnuit">Are your values generated by ignoring the differences between the rich and the poor? Are you in a position to benefit from the differences between them? Are your values chosen and determined by the significance of the construction that you manage? Does the background that you use to place a bid fit in the price of the consumption value you attribute to yourself? What is the reason to force the terms and the outcome of a negotiation that does not lead to the expected results?</p>`

This is the part of Python code with the settings I made for FPDF. Maybe this line must be change? If yes, How?

**pdf.cell(100, 100, txt = content[0], ln = i, align = 'L')**

**The code here:**

```python
from fpdf import FPDF

pdf = FPDF('P', 'mm', 'A4')
pdf.add_page()
pdf.set_font("Arial", size = 9)

# paragraphs
par_regex = re.compile('<p class="text_obisnuit.*?">.*?</p>')
pars = re.findall(par_regex, articol)
pars_text = list()

if (len(pars) == 0):
    print("Nu am gasit -- paragrafe text_obisnuit -- in fisierul --- {} ---.".format(file_path))
else:
    for i in range(0, len(pars)):
        if ('<p class="text_obisnuit">' in pars[i]):
            content = re.findall('<p class="text_obisnuit">(.*?)</p>', pars[i])
            if (len(content) == 0):
                print("Nu am gasit text in paragraful {}, fisierul {}.".format(pars[i], file_path))
            else:
                pars_text.append(content[0])
                pdf.set_font("Arial", size = 9)
                pdf.cell(100, 100, txt = content[0], ln = i, align = 'L')
        elif ('<p class="text_obisnuit2">' in pars[i]):
            content = re.findall('<p class="text_obisnuit2">(.*?)</p>', pars[i])
            if (len(content) == 0):
                print("Nu am gasit text in paragraful {}, fisierul {}.".format(pars[i], file_path))
            else:
                pars_text.append(content[0])
                pdf.set_font("Arial", 'B', size = 9)
                pdf.cell(100, 100, txt = content[0], ln = i, align = 'L')
        else:
            continue
```

**Comment by gmischler** [comment]
Hello @me-suzy ,

if you check the [documentation](https://pyfpdf.github.io/fpdf2/index.html), then you will find that in several places it is not made very explicit, but `.cell()` does indeed only render one single line by design.
In the table of the various text methods in [Adding Text](https://pyfpdf.github.io/fpdf2/Text.html), it is marked with "one" in the column "lines" (phew, at least there's a hint somewhere!).
In the section about [cell()](https://pyfpdf.github.io/fpdf2/Text.html#cell) on the same page, as well as in the [cell() API documentation](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.cell), it just mentions that it creates a rectangular area containing text. It doesn't mention line breaks either though. Apparently you're supposed to deduce from that that you can't expect any of those... :wink:
I guess that would qualify as a documentation bug, which is clearly worth to be fixed.

But back to your problem and the actual solution:
If you use `.multi_cell()`, then it will do exactly what you expected `.cell()` to do. 
* [multi_cell()  general description](https://pyfpdf.github.io/fpdf2/Text.html#multi_cell)
* [multi_cell() API documentation](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.multi_cell)

In other words, the "multi" part of its name implicitly stands for "multi-line".
Note that the `h` parameter determines the height of each individual line, and not the complete box (another point that the documentation could make more explicit).

**Comment by me-suzy** [comment]
https://pyfpdf.readthedocs.io/en/latest/reference/multi_cell/index.html

**Comment by gmischler** [comment]
> https://pyfpdf.readthedocs.io/en/latest/reference/multi_cell/index.html

@me-suzy , that is a different software package (even if a quite similar one).

For fpdf2, please refer to the documentation I've linked above.

**Comment by me-suzy** [comment]
I resolve the problem. This was the solution:
```

                                # we identify the class text_obisnuit2 and retrieve the text
                                content = re.findall('<p class="text_obisnuit2">(.*?)</p>', pars[i])
                                if (len(content) == 0):
                                    print("I did not find any text in the paragraph {}, fisierul {}.".format(pars[i], file_path))
                                else:

                                    # I set the font to bold
                                    pdf.set_font('helvetica', size=12, style="B")

                                    # we put the text in a multi_cell
                                    for simbol in dict_simboluri.keys():
                                        content[0] = content[0].replace(simbol, dict_simboluri[simbol])
                                    pars_text.append(content[0])
                                    pdf.multi_cell(w=190, txt = content[0])

                                    # add a blank line between paragraphs
                                    pdf.ln();

                                    # reset the font
                                    pdf.set_font('helvetica', size=12)
```

**Comment by Lucas-C** [comment]
@me-suzy : glad you found a solution!

Closing this issue then
